fastestimator/op/tensorop/model/update.py

```diff
@@ -32,35 +34,70 @@ class UpdateOp(TensorOp):
 
     Args:
         model: Model instance compiled by fe.build.
-        loss_name: The name of loss.
+        loss_name: The input loss key.
+        gradients: The gradients key which the model will update according to. The provided gradients will be directly
+            used for model update. If it is None, the gradients will be computed from the input key of `loss_name`, and
+            this will take care of the scaling for mixed-precision training. `gradients` should be None when the model
+            is mixed-precision.
         mode: What mode(s) to execute this Op in. For example, "train", "eval", "test", or "infer". To execute
             regardless of mode, pass None. To execute in all modes except for a particular one, you can pass an argument
             like "!infer" or "!train".
+        merge_grad: The gradient accumulation times before model update. Ex: if `merge_grad` = 3, for every three Op
+            calls only the third one updates the model. The first two calls only accumulate its gradients. This default
+            value is 1 and it will update the model at every step.
         defer: Whether to defer the actual application of the update until the end of the step. This can be necessary
             in PyTorch when trying to update multiple models which depend on one another (ex. certain GANs). By default,
             all UpdateOps which appear contiguously as the last ops of a Network will be deferred. We hope that you will
             never need to worry about this flag, but it's here for you if you need it.
+
+    Raise:
+        ValueError: When model is mixed-precision and `gradients` is provided.
+        ValueError: Network framework is not one of "tf" or "torch".
+        RuntimeError: If attempting to modify a PyTorch model which relied on gradients within a different PyTorch model
+            which has in turn already undergone a non-deferred update.
     """
     def __init__(self,
                  model: Union[tf.keras.Model, torch.nn.Module],
                  loss_name: str,
                  gradients: Optional[str] = None,
                  mode: Union[None, str, Iterable[str]] = "train",
+                 merge_grad: int = 1,
                  defer: bool = False):
+        if gradients is None:
+            super().__init__(inputs=loss_name, outputs=None, mode=mode)
+        elif model.mixed_precision:
+            raise ValueError("Mixed precision training cannot take input gradients, because the gradients need to be "
+                             "computed in this module")
+        else:
+            super().__init__(inputs=gradients, outputs=None, mode=mode)
+
+        if not hasattr(model, "loss_name"):
+            model.loss_name = {loss_name}
+        else:
+            model.loss_name.add(loss_name)
+
         self.model = model
         self.retain_graph = False
         self.weight_decay = isinstance(self.model, tf.keras.Model) and self.model.losses
```

Comment:
what's the deal with weight decay? does pytorch do this automatically?

Reply:
It was there before added by @vbvg2008